### PR TITLE
[mellanox] syncd and pmon support for debian 11

### PR DESCRIPTION
**Pull request by alexrallen**
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: platform/mellanox/docker-saiserver-mlnx.mk [hunk 1]
```diff
@@ -22,12 +22,10 @@ $(DOCKER_SAISERVER_MLNX)_DEPENDS += $(SAISERVER) $(PYTHON_SDK_API)
 $(DOCKER_SAISERVER_MLNX)_PYTHON_DEBS += $(MLNX_SFPD)
 $(DOCKER_SAISERVER_MLNX)_LOAD_DOCKERS += $(DOCKER_CONFIG_ENGINE_STRETCH)
 SONIC_DOCKER_IMAGES += $(DOCKER_SAISERVER_MLNX)
-SONIC_STRETCH_DOCKERS += $(DOCKER_SAISERVER_MLNX)
```
Comment (dgsudharsan) [inline platform/mellanox/docker-saiserver-mlnx.mk:25]: I am just curious why we had stretch target when we migrated to buster. Are you aware?

File: platform/mellanox/issu-version/Makefile [hunk 2]
```diff
@@ -19,15 +19,10 @@ SHELL = /bin/bash
 .SHELLFLAGS += -e
 
 MAIN_TARGET = issu-version
-BUSTER_FILES_PATH = /sonic/target/files/buster
 
 $(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
-ifneq (,$(findstring buster,$(DEST)))
 	sx_sdk --version | egrep -o 'ISSU [0-9]+' > $(DEST)/$(MAIN_TARGET) || {
```
Comment (dgsudharsan) [inline platform/mellanox/issu-version/Makefile:24]: Can you please fix the indentation?

File: rules/docker-platform-monitor.mk [hunk 3]
```diff
@@ -61,5 +61,5 @@ $(DOCKER_PLATFORM_MONITOR)_BASE_IMAGE_FILES += cmd_wrapper:/usr/sbin/SmartCmd
 $(DOCKER_PLATFORM_MONITOR)_BASE_IMAGE_FILES += cmd_wrapper:/usr/bin/ethtool
 $(DOCKER_PLATFORM_MONITOR)_FILES += $(SUPERVISOR_PROC_EXIT_LISTENER_SCRIPT)
 
-SONIC_BUSTER_DOCKERS += $(DOCKER_PLATFORM_MONITOR)
-SONIC_BUSTER_DBG_DOCKERS += $(DOCKER_PLATFORM_MONITOR_DBG)
+SONIC_BULLSEYE_DOCKERS += $(DOCKER_PLATFORM_MONITOR)
```
Comment (dgsudharsan) [inline rules/docker-platform-monitor.mk:64]: By moving pmon to bullseye isn't it required for all vendors to support bullseye? Unlike syncd pmon is common right?

File: sonic-slave-bullseye/Dockerfile.j2 [hunk 4]
```diff
@@ -62,6 +62,8 @@ RUN apt-get update && apt-get install -y \
         dh-exec \
         kmod \
         libtinyxml2-dev \
+	python \
```
Comment (dgsudharsan) [inline sonic-slave-bullseye/Dockerfile.j2:65]: This is for SDK compilation correct?

File: platform/mellanox/docker-syncd-mlnx.mk [hunk 5]
```diff
@@ -17,7 +17,39 @@
 # docker image for mlnx syncd
 
 DOCKER_SYNCD_PLATFORM_CODE = mlnx
-include $(PLATFORM_PATH)/../template/docker-syncd-base.mk
+
```
Comment (dgsudharsan) [inline platform/mellanox/docker-syncd-mlnx.mk:20]: Please move this section to the template and have if condition for bullseye there.

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: platform/mellanox/docker-saiserver-mlnx.mk (hunk 1 above)
Comment (alexrallen) [reply platform/mellanox/docker-saiserver-mlnx.mk:25]: Not sure, my impression is that saiserver is not in use so it probably didn't throw any errors.

File: platform/mellanox/issu-version/Makefile (hunk 2 above)
Comment (alexrallen) [reply platform/mellanox/issu-version/Makefile:24]: I believe the indentation here is correct. Under a given target we indent 4 spaces to show the code that is being executed for that target. 

When I added the ifneq I added it and didn't change the indentation.

File: rules/docker-platform-monitor.mk (hunk 3 above)
Comment (alexrallen) [reply rules/docker-platform-monitor.mk:64]: Not exactly... all vendors platform API must support python 3 is the only restriction that I am aware of.

File: sonic-slave-bullseye/Dockerfile.j2 (hunk 4 above)
Comment (alexrallen) [reply sonic-slave-bullseye/Dockerfile.j2:65]: Correct. Looks like I need to fix an indention here.

File: platform/mellanox/docker-syncd-mlnx.mk (hunk 5 above)
Comment (alexrallen) [reply platform/mellanox/docker-syncd-mlnx.mk:20]: I am not able to do that I don't think. 

This logic depends on whether a given platform's syncd supports bullseye, not if the current build environment is bullseye (I'm not sure that we have that variable at this point in the build process) 

This code itself is setting up that dependency here. We need to run the code that depends syncd on bullseye for our code and only our code.

The appropriate thing to do that would most closely mirror your suggestion is to add an if statement checking what platform we are building for but I figured we did not want such logic in the template.

Comment (dgsudharsan) [reply platform/mellanox/docker-syncd-mlnx.mk:20]: Understood. Instead can we create docker-syncd-base-bullseye.mk and have this as common code there? Later when all vendors are migrated we can ask MSFT to move it docker-syncd-base.mk

Comment (alexrallen) [reply platform/mellanox/docker-syncd-mlnx.mk:20]: I see. I think this will work fine.

**Review by dgsudharsan — APPROVED** [review]

File: platform/mellanox/docker-saiserver-mlnx.mk (hunk 1 above)
Comment (alexrallen) [reply platform/mellanox/docker-saiserver-mlnx.mk:25]: Got proven wrong by the RPC build here...

**Pull request closed without merging**